plugins/verifier/vulnerabilityreport/vulnerability_report.go

```diff
@@ -0,0 +1,400 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"embed"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/deislabs/ratify/pkg/common"
+	"github.com/deislabs/ratify/pkg/ocispecs"
+	"github.com/deislabs/ratify/pkg/referrerstore"
+	_ "github.com/deislabs/ratify/pkg/referrerstore/oras"
+	"github.com/deislabs/ratify/pkg/verifier"
+	"github.com/deislabs/ratify/pkg/verifier/plugin/skel"
+	"github.com/deislabs/ratify/plugins/verifier/vulnerabilityreport/schemavalidation"
+	imagespec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/owenrumney/go-sarif/v2/sarif"
+)
+
+//go:embed schemavalidation/schemas
+var embeddedFS embed.FS
+
+const (
+	SarifMediaType       string = "application/sarif+json"
+	SarifOfflineFilePath string = "schemavalidation/schemas/sarif-2.1.0.json"
+	TrivyScannerName     string = "trivy"
+	GrypeScannerName     string = "grype"
+	CreatedAnnotation    string = imagespec.AnnotationCreated
+	SeverityRegex               = `Severity:\s*(\w+)`
+)
+
+type PluginConfig struct {
+	Name               string   `json:"name"`
+	SchemaURL          string   `json:"schemaURL,omitempty"`
+	MaximumAge         string   `json:"maximumAge,omitempty"`
+	DisallowedSeverity []string `json:"disallowedSeverity,omitempty"`
+	Passthrough        bool     `json:"passthrough,omitempty"`
+	DenylistCVEs       []string `json:"denylistCVEs,omitempty"`
+}
+
+type PluginInputConfig struct {
+	Config PluginConfig `json:"config"`
+}
+
+func main() {
+	skel.PluginMain("vulnerabilityreport", "1.0.0", VerifyReference, []string{"1.0.0"})
```

Comment:
For the SBOM verifier, i m considering releasing with a alpha version ( which is a problem with hardcoded 1.0.0 version in the CRD controller). More discussion on versioning needed..